src/metaswitch/clearwater/queue_manager/test/test_add_to_queue.py

```diff
@@ -0,0 +1,180 @@
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+import unittest
+from .test_base import BaseQueueTest
+from metaswitch.clearwater.etcd_shared.test.mock_python_etcd import EtcdFactory
+from metaswitch.clearwater.queue_manager.etcd_synchronizer import EtcdSynchronizer
+from .plugin import TestPlugin
+from mock import patch, MagicMock
+from threading import Thread
+from time import sleep
+import json
+
+alarms_patch = patch("metaswitch.clearwater.queue_manager.alarms.issue_alarm", new=MagicMock)
+
+class AddToQueueTest(BaseQueueTest):
+    @patch("etcd.Client", new=EtcdFactory)
+    def setUp(self):
+        alarms_patch.start()
+        self._p = TestPlugin()
+        self._e = EtcdSynchronizer(self._p, "10.0.0.1", "local", "clearwater", "node")
+        self._e.WAIT_FOR_TIMER_POP = 0
+
+    @patch("etcd.Client", new=EtcdFactory)
+    def test_add_to_empty_queue(self):
+        self.set_initial_val("{\"FORCE\": false, \"ERRORED\": [], \"COMPLETED\": [], \"QUEUED\": []}")
+        self._e.add_to_queue()
+
+        val = json.loads(self._e._client.read("/clearwater/local/configuration/queue_test").value)
+        self.assertEqual(0, len(val.get("ERRORED")))
+        self.assertEqual(0, len(val.get("COMPLETED")))
+        self.assertEqual(1, len(val.get("QUEUED")))
+        self.assertEqual("10.0.0.1-node", val.get("QUEUED")[0]["ID"])
+
+    @patch("etcd.Client", new=EtcdFactory)
+    def test_add_to_queue_already_processing(self):
+        self.set_initial_val("{\"FORCE\": false, \"ERRORED\": [], \"COMPLETED\": [], \"QUEUED\": [{\"ID\":\"10.0.0.1-node\",\"STATUS\":\"PROCESSING\"}]}")
+        self._e.add_to_queue()
+
+        val = json.loads(self._e._client.read("/clearwater/local/configuration/queue_test").value)
+        self.assertEqual(0, len(val.get("ERRORED")))
+        self.assertEqual(0, len(val.get("COMPLETED")))
+        self.assertEqual(2, len(val.get("QUEUED")))
+        self.assertEqual("10.0.0.1-node", val.get("QUEUED")[0]["ID"])
+        self.assertEqual("10.0.0.1-node", val.get("QUEUED")[1]["ID"])
+        self.assertEqual("QUEUED", val.get("QUEUED")[1]["STATUS"])
+
+    @patch("etcd.Client", new=EtcdFactory)
+    def test_add_to_queue_already_queued(self):
+        self.set_initial_val("{\"FORCE\": false, \"ERRORED\": [], \"COMPLETED\": [], \"QUEUED\": [{\"ID\":\"10.0.0.1-node\",\"STATUS\":\"PROCESSING\"}, {\"ID\":\"10.0.0.1-node\",\"STATUS\":\"QUEUED\"}]}")
+        self._e.add_to_queue()
+
+        val = json.loads(self._e._client.read("/clearwater/local/configuration/queue_test").value)
+        self.assertEqual(0, len(val.get("ERRORED")))
+        self.assertEqual(0, len(val.get("COMPLETED")))
+        self.assertEqual(2, len(val.get("QUEUED")))
+        self.assertEqual("10.0.0.1-node", val.get("QUEUED")[0]["ID"])
```

Comment:
Having a Queue object with human-understandable methods might help the tests a bit too.